packages/components/inputs/rich-text-utils/src/html/html.tsx

```diff
@@ -179,12 +179,16 @@ const wrapWithParagraph = (
   textContent: TElement | TText | (TElement | TText)[]
 ) => jsx('element', { type: 'paragraph' }, textContent);
 
-const wrapWithParagraphIfRootElement = (
+const wrapWithFragment = (
+  textContent: TElement | TText | (TElement | TText)[]
+) => jsx('fragment', {}, textContent);
+
+const wrapWithFragmentIfRootElement = (
   el: HTMLElement | ChildNode,
   textContent: TElement | TText | (TElement | TText)[]
 ) =>
   el.parentNode?.nodeName === 'BODY' // root element, because body is eventually turned to React fragment
-    ? wrapWithParagraph(textContent)
+    ? wrapWithFragment(textContent)
```

Comment:
Would this be a problem in case of new line? 

In the below screenshot, there are two individual elements and above condition would be true since the parentNode is 'BODY' in this case, isn't it?

This would make the text to be in the single line without any space between them if the `<p>` tag is removed. 

<img width="1259" alt="Screenshot 2023-05-16 at 10 47 04" src="https://github.com/commercetools/ui-kit/assets/83634106/5b7d80f4-ba8a-4c44-ae8f-a57a62bb9a33">

Reply:
Thanks,
I'm not sure I 100% understand the question but let me try to answer.
You're right that in this case we'll have:
```
Body                                          
  |                                            
  |                                            
  |----P                                       
  |    |                                       
  |    |                                       
  |    |                                       
  |    +------ #text (this is first line)                        
  |                                            
  |                                            
  +-----P                                      
        |                                      
        |                                      
        +------#text (this is second line)                           
```
This function only replaces `<body>` with `<></>`.
Therefore, if you provide:
```tsx
<LocalizedRichTextInput
        onChange={() => {}}
        value={{
          en: '<p>this is the first line</p><p>this is the second line</p>',
        }}
        selectedLanguage="en"
        horizontalConstraint={7}
        defaultExpandMultilineText={true}
      />
```
the wrapping paragraphs are there.
I don't get the part how do you expect the `<p>`s to be removed? They will always be there if this content is typed in the input by the user.

Reply:
Thanks for the detailed explanation. I understand what you mean. 
However I am confused since you linked this [issue](https://github.com/commercetools/ui-kit/issues/2522) which talks about removing `<p>` tags from unformatted text content. So does this PR removes `<p>` only if there is only one `<p>` tag at the top level?  

This was my overall assumption. 
If users type anything in rich text editor without formatting (i.e, not clicking bold or italic, etc) then text will not be wrapped by any elements like `<p>`. 
If the users type anything and format their content, then respective HTML elements will be wrapped.
Is my understanding or expectations correct or is this something not possible to achieve it?

Reply:
Thanks again,
> If users type anything in rich text editor without formatting (i.e, not clicking bold or italic, etc) then text will not be wrapped by any elements like <p>.

Got it. Not sure, I need to check that.

Reply:
> If users type anything in rich text editor without formatting (i.e, not clicking bold or italic, etc) then text will not be wrapped by any elements like `<p>`.
> If the users type anything and format their content, then respective HTML elements will be wrapped.
> Is my understanding or expectations correct or is this something not possible to achieve it?

To my understanding, a richt-text input always emits HTML and always wraps the content into an element by default. For instance, typing just some text is considered a paragraph thus wrapping it into a `<p>` is in theory correct.

Reply:
I’m not sure about this one, but since the component is formatting text, I believe it needs to output some kind of rich information where we don’t only get the content but also its structure and styles. I mean, we need to know if a text is part of a list, if it’s a header, etc.

For what I can see in other libraries, the output uses to be HTML (which includes content, structure and styles) or JSON where all of those can be represented ([example](https://editorjs.io/)).

If the first element in the output is a list, I guess we wouldn’t be questioning if we want it to be wrapped in a `ul/ol`.

I’m not sure I understand the underlying problem.